### PR TITLE
[Th2 5118] read-db doesn't store results from init query in puling mode

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

File: core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/DataBaseMonitorServiceImpl.kt [hunk 1]
```diff
@@ -101,7 +101,9 @@ class DataBaseMonitorServiceImpl(
             dataSourceId,
             initQueryId,
             initParameters,
-        ).lastOrNull()
+        ).onEach {
+            updateListener.onUpdate(dataSourceId, it)
+        }.lastOrNull()
```
Comment (OptimumCode) [inline core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/DataBaseMonitorServiceImpl.kt:106]: This is not compatible with the old behavior. Believe we should introduce a parameter for the task and depending on the parameter's value call this method or don't

Comment (Nikita-Smirnov-Exactpro) [reply core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/DataBaseMonitorServiceImpl.kt:106]: Maybe I wrong and we can configure base read-db version to start pulling from empty init query result

File: README.md [hunk 2]
```diff
@@ -87,8 +87,9 @@ The read tasks tries to read all data from the specified data source using speci
 Pulls updates from the specified data source using the specified queries.
 
 + dataSource - the id of the source that should be used
-+ initQueryId - the id of the query that should be used to retrieve the current state of the database
-+ initParameters - the parameters that should be used in the init query
++ initQueryId - the id of the query that should be used to retrieve the current state of the database.
+  NOTE: this parameter is used to initialize state and read-db doesn't publish retrieved messages to MQ router.
++ initParameters - the parameters that should be used in the init query. Also, The task use these parameters to configure the fist `updateQuery` if `initQuery` parameter is missing
```
Comment (OptimumCode) [inline README.md:92]:
```suggestion
+ initParameters - the parameters that should be used in the init query. Also, The task uses these parameters to configure the first `updateQuery` execution if `initQuery` parameter is not specified
```

File: README.md [hunk 3]
```diff
@@ -177,6 +178,12 @@ spec:
 
 ## Changes
 
+### 0.3.3
+
+#### Changed:
+
++ `initQuery` parameter in pull task is optional
```
Comment (OptimumCode) [inline README.md:185]:
```suggestion
+ `initQuery` parameter in a pull task is made optional
```

File: core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/DataBaseMonitorServiceImpl.kt [hunk 4]
```diff
@@ -152,4 +154,4 @@ class DataBaseMonitorServiceImpl(
     }
 }
 
-private class TaskHolder(val job: Job, val updateListener: UpdateListener)
+private class TaskHolder(val job: Job)
```
Comment (OptimumCode) [inline core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/DataBaseMonitorServiceImpl.kt:157]: Please, roll back this change. Maybe it is unused for now but the task consists of the pulling job and listener for this job. And I want to keep those entities together.

File: core/src/main/kotlin/com/exactpro/th2/read/db/app/DataBaseReaderConfiguration.kt [hunk 5]
```diff
@@ -56,7 +56,7 @@ data class ReadTaskConfiguration(
 @JsonTypeName("pull")
 data class PullTaskConfiguration(
     val dataSource: DataSourceId,
-    val initQueryId: QueryId,
+    val initQueryId: QueryId?,
```
Comment (OptimumCode) [inline core/src/main/kotlin/com/exactpro/th2/read/db/app/DataBaseReaderConfiguration.kt:59]: Please, modify the gRPC implementation as well. It should correctly process the pulling task request without init query set

File: core/src/main/kotlin/com/exactpro/th2/read/db/app/Requests.kt [hunk 6]
```diff
@@ -29,7 +29,7 @@ class ExecuteQueryRequest(
 
 class PullTableRequest(
     val dataSourceId: DataSourceId,
-    val initQueryId: QueryId,
+    val initQueryId: QueryId?,
```
Comment (OptimumCode) [inline core/src/main/kotlin/com/exactpro/th2/read/db/app/Requests.kt:32]: Please, modify the gRPC implementation as well. It should correctly process the pulling task request without init query set

**Review by OptimumCode — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/DataBaseMonitorServiceImpl.kt [hunk 7]
```diff
@@ -89,19 +89,21 @@ class DataBaseMonitorServiceImpl(
...
     ) {
-        val lastRow: TableRow? = dataBaseService.executeQuery(
-            dataSourceId,
-            initQueryId,
-            initParameters,
-        ).lastOrNull()
+        val lastRow: TableRow? = initQueryId?.let {
+            dataBaseService.executeQuery(
+                dataSourceId,
+                initQueryId,
```
Comment (OptimumCode) [inline core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/DataBaseMonitorServiceImpl.kt:103]: It is better to use `it` here (or give `it` a name in lambda block)

File: grpc/proto/th2_grpc_read_db/read_db.proto [hunk 8]
```diff
@@ -43,7 +43,7 @@ message AssociatedMessageType {
 
 message DbPullRequest {
   SourceId source_id = 1;
-  QueryId init_query_id = 2;
+  optional QueryId init_query_id = 2;
```
Comment (OptimumCode) [inline grpc/proto/th2_grpc_read_db/read_db.proto:46]: This is proto3. All fields are optional here. I don' think we need to change this.
But if you want to change it you also need to update the gRPC version

**Review by OptimumCode — CHANGES_REQUESTED** [review]

**Review by OptimumCode — APPROVED** [review]

**Pull request merged**